docs/docs/tutorial/creating-a-source-plugin/part-6/index.mdx

```diff
@@ -0,0 +1,621 @@
+---
+title: "Part 6: Image CDN"
+---
+
+import { LinkButton } from "gatsby-interface"
+import Collapsible from "@components/collapsible"
+import { MdArrowForward } from "react-icons/md"
+
+## Introduction
+
+Gatsby always had best-in-class image support with its [`gatsby-plugin-image`](/plugins/gatsby-plugin-image/) feature. With it you can easily add responsive, optimized images to sites in all modern formats. Traditionally, images were downloaded and transformed to enable this behavior. Image CDN solves this by skipping image processing during the build, and instead deferring and offloading it to a dedicated image CDN. Visitors still get images in the most optimized formats such as AVIF and WebP, but site builds complete in a fraction of the time.
+
+By the end of this part of the tutorial, you will be able to:
+
+- Add Image CDN support to your source plugin
+- Use Image CDN with `gatsby-plugin-image` in your example site
+
+## Add Image CDN to your plugin
+
+Gatsby's Image CDN feature can be added to any Gatsby source plugin. We'd recommend using it over manual file downloading for the reasons above. As long as your API gives back an URL to an image, you should be able to implement the required code. Basically, for Image CDN to work you need two things:
+
+- A GraphQL root node type that implements the `RemoteFile` interface
+- During node creation for that GraphQL node type, certain fields must exist on the node to match what `RemoteFile` expects
+
+In [Part 3](/docs/tutorial/creating-a-source-plugin/part-3/#createtypes) you learned that root node types need to implement the `Node` interface. Similarly to `Node`, `RemoteFile` is also provided by Gatsby itself and will handle all the complicated pieces of Image CDN behind the scenes.
+
+<Announcement>
+
+**Pro tip:** Gatsby's Image CDN feature is smart enough to work on all platforms, even if no CDN provider like on Gatsby Cloud or Netlify is available. In those cases (including locally on your computer) it automatically falls back to processing images during the build. This won't give users a build improvement but things on the frontend will still behave the same and builds won't fail.
+
+</Announcement>
+
+The `RemoteFile` interface has the following shape:
+
+```graphql
+interface RemoteFile {
+  id: ID!
+  mimeType: String!
+  filename: String!
+  filesize: Int
+  width: Int
+  height: Int
+  publicUrl: String!
+  resize(/* args */): RemoteFileResize
+  gatsbyImage(/* args */): GatsbyImageData
+}
+```
+
+For node types implementing `RemoteFile` this means:
+
+- Always required are `publicUrl` (the url to the image/file), [`mimeType`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_Types) (e.g. `image/jpg` or `application/pdf`), and `filename`. `publicUrl` will be defined through `url` during node creation.
+- `width`, `height`, `resize`, and `gatsbyImage` can be null. This is because the `RemoteFile` interface can also handle assets other than images, like PDFs
```

Comment:
```suggestion
- `width`, `height`, `resize`, and `gatsbyImage` can be null. This is because the `RemoteFile` interface can also handle assets other than images, like PDFs.
```